[TEST-FAILURE] ReadWriteThroughJCacheTests.test_loadall_readthrough

```
java.lang.AssertionError: expected:<100> but was:<98>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.cache.ReadWriteThroughJCacheTests.test_loadall_readthrough(ReadWriteThroughJCacheTests.java:157)
```
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OracleJDK1.6/com.hazelcast$hazelcast/499/testReport/junit/com.hazelcast.cache/ReadWriteThroughJCacheTests/test_loadall_readthrough/

--- a/hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheClearOperation.java
+++ b/hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheClearOperation.java
@@ -32,7 +32,6 @@ public class CacheClearOperation
         extends PartitionWideCacheOperation
         implements BackupAwareOperation {
 
-
     private transient ICacheRecordStore cache;
 
     public CacheClearOperation() {

--- a/hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheLoadAllOperation.java
+++ b/hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheLoadAllOperation.java
@@ -21,6 +21,8 @@ import com.hazelcast.cache.impl.CacheDataSerializerHook;
 import com.hazelcast.cache.impl.CacheService;
 import com.hazelcast.cache.impl.ICacheRecordStore;
 import com.hazelcast.cache.impl.record.CacheRecord;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.nio.serialization.Data;
 import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
 import com.hazelcast.partition.InternalPartitionService;
@@ -30,6 +32,7 @@ import com.hazelcast.spi.PartitionAwareOperation;
 import com.hazelcast.spi.impl.AbstractNamedOperation;
 
 import javax.cache.CacheException;
+import java.io.IOException;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
@@ -47,7 +50,6 @@ public class CacheLoadAllOperation
 
     private Set<Data> keys;
     private boolean replaceExistingValues;
-
     private boolean shouldBackup;
 
     private transient Map<Data, CacheRecord> backupRecords;
@@ -78,9 +80,11 @@ public class CacheLoadAllOperation
                 }
             }
         }
+
         if (filteredKeys.isEmpty()) {
             return;
         }
+
         try {
             final CacheService service = getService();
             cache = service.getOrCreateCache(name, partitionId);
@@ -95,7 +99,6 @@ public class CacheLoadAllOperation
         } catch (CacheException e) {
             response = new CacheClearResponse(e);
         }
-
     }
 
     @Override
@@ -131,5 +134,33 @@ public class CacheLoadAllOperation
     @Override
     public final int getAsyncBackupCount() {
         return cache != null ? cache.getConfig().getAsyncBackupCount() : 0;
+    }
+
+    @Override
+    protected void writeInternal(ObjectDataOutput out) throws IOException {
+        super.writeInternal(out);
+        out.writeBoolean(replaceExistingValues);
+        out.writeBoolean(keys != null);
+        if (keys != null) {
+            out.writeInt(keys.size());
+            for (Data key : keys) {
+                out.writeData(key);
+            }
+        }
+    }
+
+    @Override
+    protected void readInternal(ObjectDataInput in) throws IOException {
+        super.readInternal(in);
+        replaceExistingValues = in.readBoolean();
+        boolean isKeysNotNull = in.readBoolean();
+        if (isKeysNotNull) {
+            int size = in.readInt();
+            keys = new HashSet<Data>(size);
+            for (int i = 0; i < size; i++) {
+                Data key = in.readData();
+                keys.add(key);
+            }
+        }
     }
 }

--- a/hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheRemoveAllOperation.java
+++ b/hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheRemoveAllOperation.java
@@ -4,12 +4,15 @@ import com.hazelcast.cache.impl.CacheClearResponse;
 import com.hazelcast.cache.impl.CacheDataSerializerHook;
 import com.hazelcast.cache.impl.ICacheRecordStore;
 import com.hazelcast.cache.impl.ICacheService;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.nio.serialization.Data;
 import com.hazelcast.partition.InternalPartitionService;
 import com.hazelcast.spi.BackupAwareOperation;
 import com.hazelcast.spi.Operation;
 
 import javax.cache.CacheException;
+import java.io.IOException;
 import java.util.HashSet;
 import java.util.Set;
 
@@ -19,11 +22,9 @@ import java.util.Set;
 public class CacheRemoveAllOperation extends PartitionWideCacheOperation implements BackupAwareOperation {
 
     private Set<Data> keys;
-
     private int completionId;
 
     private transient Set<Data> filteredKeys = new HashSet<Data>();
-
     private transient ICacheRecordStore cache;
 
     public CacheRemoveAllOperation() {
@@ -98,4 +99,31 @@ public class CacheRemoveAllOperation extends PartitionWideCacheOperation impleme
         }
     }
 
+    @Override
+    protected void writeInternal(ObjectDataOutput out) throws IOException {
+        super.writeInternal(out);
+        out.writeInt(completionId);
+        out.writeBoolean(keys != null);
+        if (keys != null) {
+            out.writeInt(keys.size());
+            for (Data key : keys) {
+                out.writeData(key);
+            }
+        }
+    }
+
+    @Override
+    protected void readInternal(ObjectDataInput in) throws IOException {
+        super.readInternal(in);
+        completionId = in.readInt();
+        boolean isKeysNotNull = in.readBoolean();
+        if (isKeysNotNull) {
+            int size = in.readInt();
+            keys = new HashSet<Data>(size);
+            for (int i = 0; i < size; i++) {
+                Data key = in.readData();
+                keys.add(key);
+            }
+        }
+    }
 }

